src/sync/pull.rs

```diff
@@ -59,8 +60,18 @@ pub async fn begin_pull(
     let pull_timer = rlog::Timer::new().map_err(InternalTimerError)?;
     let pull_resp = puller
         .pull(&pull_req, &pull_url, &pull_auth, &request_id)
-        .await
-        .map_err(PullFailed)?;
+        .await;
+    if let Err(PullError::FetchNotOk(status_code, body)) = pull_resp {
+        return Ok(BeginTryPullResponse {
```

Comment:
I would have thought this would still be an Err? Is it because we can't return the httpinfo in an error? Can we not have a(nother, lame) special case error that js recognizes?